syllabus/core/stat_recorder.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+from syllabus.task_space import TaskSpace
+from gymnasium.spaces import Discrete #, MultiDiscrete?
+import json
+import os
+
+def convert_numpy(obj):
+    if isinstance(obj, np.generic):
+        return obj.item()  # Use .item() to convert numpy types to native Python types
+    raise TypeError
+
+class StatRecorder:
+    """
+    Individual stat tracking for each task.
+    """
+
+    def __init__(self, task_space: TaskSpace):
+        """Initialize the StatRecorder"""
+
+        self.write_path = '/Users/allisonyang/Downloads'
```

Comment:
Make this an initializer argument and let people configure it when they create their curriculum